addons/mass_mailing/static/src/js/mass_mailing_html_field.js

```diff
@@ -0,0 +1,558 @@
+/** @odoo-module **/
+
+import legacyEnv from 'web.commonEnv';
+import { registry } from "@web/core/registry";
+import { _lt } from "@web/core/l10n/translation";
+import { standardFieldProps } from "@web/views/fields/standard_field_props";
+import { initializeDesignTabCss } from "mass_mailing.design_constants";
+import { toInline } from "web_editor.convertInline";
+import { loadLibs, loadXML } from "web.ajax";
+import { qweb } from 'web.core';
+import { useService } from "@web/core/utils/hooks";
+import { buildQuery } from "web.rpc";
+import { HtmlField } from "@web_editor/js/backend/html_field";
+import { getWysiwygClass } from 'web_editor.loader';
+import { device } from 'web.config';
+
+
+const {
+    useChildSubEnv,
+    useEffect,
+    onWillUpdateProps,
+} = owl;
+
+export class MassMailingHtmlField extends HtmlField {
+    setup() {
+        super.setup();
+
+        useChildSubEnv(legacyEnv);
```

Comment:
explain why legacy env

Reply:
Because there is at least one call from the legacy widget or some of it's sub-widget that require a `trigger_up` `call_service` requiring the legacy `ajax` service being present in the widget.

Note that because we also need that environment in note because the color widget also try to call the legacy `ajax` service, I moved the instruction to change the environment in the class `HtmlFieldWysiwygAdapterComponent`.